<launch>
  <arg name="vehicle" default="uav11"/>
  <arg name="rate" default="1"/>
  <arg name="index" default="1"/>
  <arg name="bagfile" default="simulation"/>
  <rosparam param="robot_list">[uav10,uav11]</rosparam>
  <rosparam param="robot_id_list">[a,b]</rosparam>

  <arg name="bagfile" default="robot_cloud"/>

  <node pkg="rosbag"
    name="rosbag_simulation_$(arg vehicle)"
    type="play"
    args="-r $(arg rate) /home/rasl/bagfiles/simulation/$(arg vehicle)_tof.bag">
  </node>


  <!-- Record important packages for replay or matlab analysis -->
  <node pkg="rosbag"
    type="record"
    name="record"
    output="screen"
    args="-O /home/rasl/bagfiles/two_robots_simulation/$(arg bagfile).bag
    /tf
    /distributed_mapping_$(arg vehicle)/knn_hist
    /distributed_mapping_$(arg vehicle)/packet_size_count
    /distributed_mapping_$(arg vehicle)/relpose_estimate
    /distributed_mapping_$(arg vehicle)/scan_points_count
    /distributed_mapping_$(arg vehicle)/match_scan_info
    /distributed_mapping_$(arg vehicle)/timer
    /cpu_monitor/cpu_status
    "/>

<!--
  <node name="rviz"
        pkg="rviz"
        type="rviz"
        output="screen"
        args="-d $(find joint_mapping)/rviz/simulation_visualization.rvix"/>
-->

  <node name="cpu_monitor"
    pkg="cpu_monitor"
    type="cpu_monitor"
    output="screen">
    <param name="time_constant" value="0.1" />
    <param name="rate" value="10" />
  </node>

  <node name="scan_odom_sync"
    pkg="scan_odom_sync"
    type="scan_odom_sync"
    output="screen">

    <param name="update_interval" value="5" />
    <param name="robot_name" value="$(arg vehicle)"/>
    <param name="robot_frame" value="$(arg vehicle)/base" />
    <param name="world_frame" value="world" />

    <remap from="~/$(arg vehicle)/point_cloud_with_pose" to="/$(arg vehicle)/point_cloud_with_pose"/>
  </node>


<!--========================= distributed_mapping nodes ========================-->
  <node pkg="joint_mapping"
    name="distributed_mapping_$(arg vehicle)"
    type="joint_mapping_node"
    output="screen">
    <!-- setting: default distributed_mapping param and agent name -->
    <rosparam file="$(find joint_mapping)/config/parameters.yaml"/>

    <!-- Index into the robot list -->
    <param name="index" value="$(arg index)"/>
    <param name="update_interval" value="5"/>

    <!-- udp lists : localhost:50000 -->
    <param name="udp/local_ip" value="192.168.10.122"/>
    <param name="udp/local_port" value="5000"/>
    <param name="udp/foreign_agent_ip1" value="192.168.10.121"/>
    <param name="udp/foreign_agent_port1" value="5000"/>

    <param name="range_max" value="15"/>

    <param name="robot_frame" value="$(arg vehicle)/base"/>

    <param name="frame_id/fixed" value="world"/>
    <param name="frame_id/dma" value="dma" />

    <remap from="~agentPackage" to="/agentPackage"/>
  </node>

  <node pkg="pcl_ros"
    type="pcd_to_pointcloud"
    name="pcd_to_pointcloud"
    output="screen"
    args="$(find joint_mapping)/maps/my_tunnel_4_noceil.pcd">
    <param name="frame_id" value="/world"/>
  </node>

</launch>

